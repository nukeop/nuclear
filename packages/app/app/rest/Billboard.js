import { getChart } from 'billboard-top-100';

export const lists = {
  genres: [
    {
      name: 'Mainstream Top 40',
      link: 'pop-songs'
    },
    {
      name: 'Adult Contemporary',
      link: 'adult-contemporary'
    },
    {
      name: 'Adult Top 40',
      link: 'adult-pop-songs'
    }
    
  ]
};

export function getTop(list) {
  return new Promise((resolve, reject) => {
    getChart(list, (songs, err) => {
      if (err) {
        reject(err);
      } else {
        resolve(songs);
      }
    });
  });
}
