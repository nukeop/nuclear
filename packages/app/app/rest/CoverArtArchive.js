const apiUrl = 'http://coverartarchive.org/';

function releaseGroupFront(group, size=250) {
  return new Promise(fulfill => {
    fetch(apiUrl + 'release-group/' + group.id + '/' + 'front-' + size)
      .then(cover => {
        if (!cover.ok) {
          group.cover = null;
          fulfill(group);
        } else {
          group.cover = cover.url;
          fulfill(group);
        }
      })
      .catch(() => {
        group.cover = null;
        fulfill(group);
      });
  });
}

module.exports = {
  releaseGroupFront
};
