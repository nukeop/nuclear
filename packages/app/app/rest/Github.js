import {
  githubClientId
} from '../globals';

const GITHUB_OAUTH_AUTHORIZE_URL = 'https://github.com/login/oauth/authorize';
export const GITHUB_OAUTH_ACCESS_TOKEN_URL = 'https://github.com/login/oauth/access_token';
export const GITHUB_API_USER_ENDPOINT = 'https://api.github.com/user';
export const GITHUB_REPO_CONTRIB_ENDPOINT = 'https://api.github.com/repos/nukeop/nuclear/stats/contributors';


export const getGithubOauthUrl = () => {
  return GITHUB_OAUTH_AUTHORIZE_URL +
    '?client_id=' +
    githubClientId +
    '&redirect_uri=' +
    window.location.href;
};
