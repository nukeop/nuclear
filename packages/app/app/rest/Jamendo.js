import globals from '../globals';

export function search (query) {
  const limit = 10;
  const url =
    'https://api.jamendo.com/v3.0/artists/tracks/' +
    '?client_id=' +
    globals.jamendoClientId +
    '&format=jsonpretty' +
    '&limit=' +
    limit +
    '&name=' +
    encodeURIComponent(query.artist) +
    '&track_name=' +
    encodeURIComponent(query.track);

  return fetch(url);
}
