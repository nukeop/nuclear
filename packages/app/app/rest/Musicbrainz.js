const NB = require('nodebrainz');
const nb = new NB({userAgent: 'Mozilla/5.0 (Windows NT 6.3; WOW64; rv:37.0) Gecko/20100101 Firefox/37.0'});
const covers = require('./CoverArtArchive');

function artistSearch(terms) {
  return new Promise((fulfill, reject) => {
    nb.search('artist', {artist: terms}, (err, response) => {
      if (err) {
        reject(err);
      } else {
        fulfill(response);
      }
    });
  });
}

function addCoversToReleases(searchResults) {
  let coverPromises = searchResults['release-groups'].map(group => {
    return covers.releaseGroupFront(group);
  });

  return Promise.all(coverPromises);
}

function releaseSearch(terms) {
  let nbSearch = new Promise((fulfill, reject) => {
    nb.search('release-group', {release: terms}, (err, response) => {
      if (err) {
        reject(err);
      } else {
        fulfill(response);
      }
    });
  });

  return nbSearch;
}

function trackSearch(terms) {
  return new Promise((fulfill, reject) => {
    nb.search('work', {work: terms}, (err, response) => {
      if (err) {
        reject(err);
      } else {
        fulfill(response);
      }
    });
  });
}

module.exports = {
  artistSearch,
  releaseSearch,
  trackSearch,
  addCoversToReleases
};
