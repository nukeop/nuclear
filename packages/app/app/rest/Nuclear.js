import logger from 'electron-timber';

const nuclearNewsUrl = 'https://nuclear.js.org/news/';

export function getNewsIndex() {
  return fetch(nuclearNewsUrl)
    .then(response => response.json())
    .then(response => {
      return Promise.resolve(response);
    })
    .catch(err => {
      logger.error(err);
    });
}

export function getNewsItem(itemName) {
  return fetch(nuclearNewsUrl + itemName)
    .then(response => response.json())
    .then(response => {
      return Promise.resolve(response);
    })
    .catch(err => {
      logger.error(err);
    });
}
