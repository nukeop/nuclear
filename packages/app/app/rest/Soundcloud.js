import globals from '../globals';
const apiUrl = 'https://api.soundcloud.com';

function prepareUrl(url) {
  return `${url}&client_id=${globals.soundcloudApiKey}`;
}

export function soundcloudSearch(terms) {
  return fetch(prepareUrl(apiUrl + '/tracks?limit=50&q=' + terms));
}
