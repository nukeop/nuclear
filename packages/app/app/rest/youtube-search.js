import { getOption } from '../persistence/store';

export function prepareUrl (url) {
  return `${url}&key=${ getOption('yt.apiKey')}`;
}

export function trackSearch (track) {
  return fetch(prepareUrl('https://www.googleapis.com/youtube/v3/search?part=id,snippet&type=video&maxResults=50&q=' + encodeURIComponent(track)));
}
