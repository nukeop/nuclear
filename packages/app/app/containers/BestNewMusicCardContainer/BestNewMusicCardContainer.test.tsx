import { waitFor } from '@testing-library/react';
import pitchforkBnm from 'pitchfork-bnm';
import { Deezer } from '@nuclear/core/src/rest';

import { buildStoreState } from '../../../test/storeBuilders';
import { mountedComponentFactory, setupI18Next } from '../../../test/testUtils';

describe('Dashboard container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  beforeEach(() => {
    const mockState = buildStoreState()
      .withDashboard()
      .build();

    jest.mock('pitchfork-bnm', () => jest.fn().mockImplementation(() => ({
      getBestNewAlbums: jest.fn().mockResolvedValue(mockState.dashboard.bestNewAlbums),
      getBestNewTracks: jest.fn().mockResolvedValue(mockState.dashboard.bestNewTracks)
    })));

    jest.mock('@nuclear/core/src/rest', () => jest.fn().mockImplementation(() => ({
      getTopTracks: jest.fn().mockResolvedValue(mockState.dashboard.bestNewAlbums),
      mapDeezerTrackToInternal: jest.requireActual('@nuclear/core/src/rest').Deezer.mapDeezerTrackToInternal
    })));

  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render card album title, album artist, track title and track artist', async () => {
    const { component } = mountComponent();
    await waitFor(() => component.getByText(/Best new music/i).click());
    expect(component.queryByTestId('best-new-music-card-title-test title 1')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-title-test title 2')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-title-test track title 1')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-title-test track title 2')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-artist-test artist 1')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-artist-test artist 2')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-artist-test track artist 1')).not.toBeNull();
    expect(component.queryByTestId('best-new-music-card-artist-test track artist 2')).not.toBeNull();
    expect(component.asFragment()).toMatchSnapshot();
  });

  const mountComponent = mountedComponentFactory(
    ['/dashboard'],
    buildStoreState()
      .withDashboard()
      .withPlugins()
      .withConnectivity()
      .withSettings({
        promotedArtists: true
      })
      .build()
  );
});
