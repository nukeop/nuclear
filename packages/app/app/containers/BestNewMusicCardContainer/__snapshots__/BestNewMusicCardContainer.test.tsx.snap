// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard container should render card album title, album artist, track title and track artist 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard_tabs"
      >
        <div
          class="ui pointing secondary menu"
        >
          <a
            class="item"
          >
            Trending
          </a>
          <a
            class="active item"
          >
            Best new music
          </a>
          <a
            class="item"
          >
            Top Tracks
          </a>
          <a
            class="item"
          >
            Genres
          </a>
        </div>
        <div
          class="ui segment active tab best_new_music_tab_pane"
        >
          <div
            class="best_new_music_menu"
          >
            <div
              class="ui large vertical menu"
            >
              <div
                class="link item best_new_music_menu_header"
              >
                Best new albums
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 1"
                      src="test thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                        data-testid="best-new-music-card-title-test title 1"
                      >
                        test title 1
                      </div>
                    </div>
                    <div
                      class="card_artist"
                      data-testid="best-new-music-card-artist-test artist 1"
                    >
                      test artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test title 2"
                      src="test thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                        data-testid="best-new-music-card-title-test title 2"
                      >
                        test title 2
                      </div>
                    </div>
                    <div
                      class="card_artist"
                      data-testid="best-new-music-card-artist-test artist 2"
                    >
                      test artist 2
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item best_new_music_menu_header"
              >
                Best new tracks
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 1"
                      src="test track thumbnail 1"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                        data-testid="best-new-music-card-title-test track title 1"
                      >
                        test track title 1
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 1-test track title 1"
                      />
                    </div>
                    <div
                      class="card_artist"
                      data-testid="best-new-music-card-artist-test track artist 1"
                    >
                      test track artist 1
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="link item"
              >
                <div
                  class="best_new_music_card"
                >
                  <div
                    class="card_thumbnail"
                  >
                    <img
                      alt="test track title 2"
                      src="test track thumbnail 2"
                    />
                  </div>
                  <div
                    class="card_info"
                  >
                    <div
                      class="card_title_row"
                    >
                      <div
                        class="card_title"
                        data-testid="best-new-music-card-title-test track title 2"
                      >
                        test track title 2
                      </div>
                      <i
                        aria-hidden="true"
                        class="star outline large icon card_favorite"
                        data-testid="favorite-icon-test track artist 2-test track title 2"
                      />
                    </div>
                    <div
                      class="card_artist"
                      data-testid="best-new-music-card-artist-test track artist 2"
                    >
                      test track artist 2
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="best_new_music_content"
          >
            <div
              class="review_header"
            >
              <div
                class="thumbnail"
              >
                <img
                  alt="test title 1"
                  src="test thumbnail 1"
                />
              </div>
              <div
                class="review_headings"
              >
                <div
                  class="artist"
                >
                  test artist 1
                </div>
                <div
                  class="title"
                >
                  test title 1
                </div>
              </div>
              <div
                class="score"
              >
                <div
                  class="score_box"
                >
                  5.0
                </div>
              </div>
            </div>
            <p
              class="paragraph"
            >
              test review 1
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
